services/bettersocial/user/BetterSocialBlockUserV2.js

```diff
@@ -58,15 +136,27 @@ const BetterSocialBlockUserV2 = async (token, selfUserId, targetUserId, source,
         {transaction: t, message, postId, reason}
       );
 
-      if (targetAnonymousUserId?.user_id)
-        await UserBlockUserFunction.userBlock(
-          UserBlockedUser,
-          UserBlockedUserHistory,
+      if (targetAnonymousUserId?.user_id) {
+        let unfollowParams = {
+          token,
+          selfAnonymousUserId,
+          targetAnonymousUserId,
+          UserFollowUser,
+          UserFollowUserHistory,
+          targetUserId,
           selfUserId,
-          targetAnonymousUserId?.user_id,
-          source,
-          {transaction: t, message, postId: '', reason, isAnonymous: true}
-        );
+          t,
+          postId
+        };
+        // Destroy follow from sign A to Sign B
+        await sign_a_unfollow_sign_b(unfollowParams);
+        // Destroy follow from sign B to sign A
+        await sign_b_unfollow_sign_a(unfollowParams);
+        // Destroy follow from sign A to Anon B
+        await sign_a_unfollow_anon_b(unfollowParams);
+        // Destroy follow from sign B to Anon A
+        await sign_b_unfollow_anon_a(unfollowParams);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `unfollowParams` object could be defined once and reused in each function call, reducing redundancy and improving readability. However, this looks fine as it is.



<!-- This is an auto-generated comment by OSS CodeRabbit -->